### PR TITLE
Move TerminalApp's resources into the TerminalApp project

### LINKED ISSUE #792: Move hard-coded strings and UI text to Resources
There are a few places in current iteration that are hard-coding UI-shown values and should move those to resources for easier updating and localization.

Need to determine if following UWP-style resources or RC based.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
This commit also introduces a scoped resource accessor, lightly taken
from microsoft-ui-xaml. It also moves all static UI strings out of
App.cpp and into localizable resources.

Fixes #792. Puts us in a better place to address #1484.

The resource accessor was necessary to make sure that TerminalApp loaded its own resources and not the toplevel resource dictionary.

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
I'm going to block on this until the TerminalApp/TerminalAppLib project dust settles (/cc @zadjii-msft)

File: src/cascadia/TerminalApp/ScopedResourceLoader.cpp [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "ScopedResourceLoader.h"
+
+using namespace ::winrt::Windows::ApplicationModel::Resources::Core;
+
+//static constexpr std::wstring_view resourceLocatorBase{ L"TerminalApp/Resources" };
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/ScopedResourceLoader.cpp]:
```suggestion
```

File: src/cascadia/TerminalApp/ScopedResourceLoader.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+
+#include "ScopedResourceLoader.h"
+
+using namespace ::winrt::Windows::ApplicationModel::Resources::Core;
+
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalApp/ScopedResourceLoader.cpp]:
```suggestion
```

**Comment by miniksa** [comment]
> I'm going to block on this until the TerminalApp/TerminalAppLib project dust settles (/cc @zadjii-msft Mike Griese FTE)

I would have merged this, but you had this comment. So I left it for you to merge when you're happy.

**Pull request merged**